tests/test_install.py

```diff
@@ -122,6 +127,48 @@ def test_calls_func(self):
         func.assert_called_with(some_path)
 
 
+@pytest.mark.parametrize("args", [
+    (),
+    ("one", ),
+    ("one", "two", "three", ),
+])
+def test_ensure_write_takes_two_arguments(args):
+    with pytest.raises(TypeError):
+        install.ensure_write(*args)
+
+
+class ensure_write_TestCase(unittest.TestCase):
+    def setUp(self):
+        self.prefix = generate_random_path()
+        self.dist = "foobar-%s-0" % random.randint(100, 200)
+        self.meta = {
+            "name": self.dist,
+            "files": ["a", "b", "c"],
+        }
+
+    @contextmanager
+    def generate_mock_check(self, return_value=True):
+        with patch.object(install, 'can_open_all_files_in_prefix') as mocked:
+            mocked.return_value = return_value
+            with patch.object(install, "load_meta") as load_meta:
+                load_meta.return_value = self.meta
+                yield mocked
+
+    def test_dispatches_to_can_open_all_files(self):
+        with self.generate_mock_check() as can:
+            install.ensure_write(self.prefix, self.dist)
+        can.assert_called_with(self.prefix, self.meta['files'])
+
+    def test_returns_none_on_true(self):
+        with self.generate_mock_check():
+            actual = install.ensure_write(self.prefix, self.meta)
```

Comment:
The approachability is another story. If you want my honest opinion on that, it's "don't use mock or unittest" :)

I'm mainly concerned with how this test was ever passing. I tried moving ensure_write to a different file and somehow that triggered it to stop working, but it seems to me like it should have never worked in the first place (this was before I decided to just remove the function).